spec/features/reset_password_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+feature "reset password" do
+  scenario "user uses email password reset" do
+    user = Fabricate(:user)
+    visit_forgot_password
+    fill_and_submit_reset
+    expect_confirmation
+    click_email_reset_link
+    fill_and_submit_new_password
+    fill_and_submit_new_login
+    expect_login
+  end
+end
+
+def visit_forgot_password
+  visit '/'
+  click_on "Sign In"
+  click_on "Forgot Password?"
+end
+
+def fill_and_submit_reset
+  fill_in "Email Address", with: User.first.email
+  click_on "Send Email"
+end
+
+def expect_confirmation
+  expect(current_path).to eq "/confirmation"
```

Comment:
I would recommend using named paths. You don't want your tests dependent on the actual url, just the route.